core/domain/state_domain.py

```diff
@@ -533,25 +608,16 @@ def get_all_html_content_strings(self):
             list(str): The list of all html content strings in the interaction.
         """
         html_list = []
+        interaction = (
+            interaction_registry.Registry.get_interaction_by_id(
+                self.id))
```

Comment:
Here self.id can be None, right?

Reply:
Yeah, It is none when a state has no interaction.
I'll add a test case for this.